src/go/k8s/internal/controller/redpanda/cluster_controller.go

```diff
@@ -1158,6 +1158,13 @@ func getQuiescentCondition(redpandaCluster *vectorizedv1alpha1.Cluster) vectoriz
 		return condition
 	}
 
+	if redpandaCluster.Status.CurrentReplicas != redpandaCluster.Status.Replicas || redpandaCluster.Status.Replicas != redpandaCluster.Status.ReadyReplicas || (redpandaCluster.Spec.Replicas != nil && *redpandaCluster.Spec.Replicas != redpandaCluster.Status.Replicas) {
+		condition.Status = corev1.ConditionFalse
+		condition.Reason = "ReplicasNotSynced"
+		condition.Message = fmt.Sprintf("Replicas are not synced. spec.replicas=%d status.currentReplicas=%d,status.replicas=%d,status.readyReplicas=%d. All must be equal.", ptr.Deref(redpandaCluster.Spec.Replicas, 0), redpandaCluster.Status.CurrentReplicas, redpandaCluster.Status.Replicas, redpandaCluster.Status.ReadyReplicas)
```

Comment:
If Replicas is wrapped in a `ptr.Deref` here shouldn't it be wrapped in the above condition as well?

Reply:
in above condition i do a nil check

Reply:
i now changed both to use ptr.Deref to treat nil as 0 here.
with NodePools, this logic has to change a bit, because nil will be a quite common and legitimate value. but it's not a problem at the moment